### PR TITLE
Rustdoc blog post

### PR DESCRIPTION
**Pull request by apiraino**
cc @GuillaumeGomez @jyn514 

This is the draft for the blog post to gather interest around the rustdoc project and attract new contributors.

This draft has already gone through a first review on [Zulip](https://rust-lang.zulipchat.com/#narrow/stream/266220-rustdoc/topic/Recruiting.20new.20rustdoc.20members).

The file name suggests a publication date for **Friday, 18th December** but it's not a deadline; entirely up to you, if you're happy with this PR or we want to further iterate over it.

As such, comments and suggestions are welcome, thanks!

<details><summary>Rendered version</summary>

dark

</details>

### CODE REVIEW

File: posts/2020-12-18-rustdoc-new-contributors.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+layout: post
+title: "Rustdoc call for new contributors"
```
Comment (apiraino) [inline posts/2020-12-18-rustdoc-new-contributors.md:3]: The title is a bit "dry", I'll think of something more appealing, inspiring ... better.

File: posts/2020-12-18-rustdoc-new-contributors.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+In order to give concrete expectations, it's worth mentioning that at its current stage the project does have a medium/high level bar to new contributors and some time might be needed to get the proper context to land your first PR.
+
+## How to get started
+
+The code for rustdoc is in [the main Rust repository](https://github.com/rust-lang/rust/tree/master/src/librustdoc). The rustc-dev-guide has an [an introduction to rustdoc](https://rustc-dev-guide.rust-lang.org/rustdoc.html), as well as a detailed dive into [how rustdoc works internally](https://rustc-dev-guide.rust-lang.org/rustdoc-internals.html).
+
+## Where to get in contact
+
+The official channel for day to day conversations will be soon on Zulip at the [#rustdoc stream](https://rust-lang.zulipchat.com/#narrow/stream/266220-rustdoc). If you happen to be on [Discord](https://discord.gg/4yEYPuT), there is also the channel there.
```
Comment (jyn514) [inline posts/2020-12-18-rustdoc-new-contributors.md:50]: https://github.com/rust-lang/team/pull/492
```suggestion
The official channel for day to day conversations is on Zulip at the [#rustdoc stream](https://rust-lang.zulipchat.com/#narrow/stream/266220-rustdoc). If you happen to be on [Discord](https://discord.gg/4yEYPuT), there is also the channel there.
```

**Review by jyn514 — COMMENTED** [review]
I would still somewhat prefer to wait for https://github.com/rust-lang/rust/pull/79540, but that's taking long enough it may not be worth it :/ my main concern is that people will see the compile times and 'bounce', decide that it's not worth contributing and never try again even after they've improved.

File: posts/2020-12-18-rustdoc-new-contributors.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+- want to get engaged in the free/open source community
+
+then this call is for you!
+
+
+## What is `rustdoc`?
+
+Rustdoc builds the documentation for a Rust crate. It ships with your Rust compiler and it is invoked with `cargo doc`.
+
+rustdoc is not to be confused with docs.rs (even though docs.rs is using rustdoc!): docs.rs is the website hosting the documentation for Rust crates and it's a separate entity with different stewardship (some contributors share their work among these two projects).
```
Comment (bjorn3) [inline posts/2020-12-18-rustdoc-new-contributors.md:25]:
```suggestion
Rustdoc is not to be confused with docs.rs (even though docs.rs is using rustdoc!): docs.rs is the website hosting the documentation for Rust crates and it's a separate entity with different stewardship (some contributors share their work among these two projects).
```

**Comment by steveklabnik** [comment]
This post should probably be on Inside Rust, not the main blog, right? The main blog is end-user focused, but Inside Rust is "about the development of the project" focused.

**Comment by apiraino** [comment]
> This post should probably be on Inside Rust, not the main blog, right? The main blog is end-user focused, but Inside Rust is "about the development of the project" focused.

uhm .. you have a point but I thought to file it under "help needed" like we did for the [WG-prioritization call for contribution](https://blog.rust-lang.org/2020/09/14/wg-prio-call-for-contributors.html) but now I'm not sure I did the right thing with the other blog post :-) In general we want to keep these kind of blog posts all on Inside Rust, if I understand, right?

**Comment by apiraino** [comment]
> I would still somewhat prefer to wait for [rust-lang/rust#79540](https://github.com/rust-lang/rust/pull/79540), but that's taking long enough it may not be worth it :/ my main concern is that people will see the compile times and 'bounce', decide that it's not worth contributing and never try again even after they've improved.

I had a cursory read at that PR and I agree. We can publish this when the rustdoc team feels is ok, no rush

File: posts/2020-12-18-rustdoc-new-contributors.md (hunk 1 above)
Comment (GuillaumeGomez) [reply posts/2020-12-18-rustdoc-new-contributors.md:3]: "You hate writing docs? So are we! Why not joining rustdoc then?" :rofl:

File: posts/2020-12-18-rustdoc-new-contributors.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+- want to get engaged in the free/open source community
+
+then this call is for you!
+
+
+## What is `rustdoc`?
+
+Rustdoc builds the documentation for a Rust crate. It ships with your Rust compiler and it is invoked with `cargo doc`.
+
+Rustdoc is not to be confused with docs.rs (even though docs.rs is using rustdoc!): docs.rs is the website hosting the documentation for Rust crates and it's a separate entity with different stewardship (some contributors share their work among these two projects).
```
Comment (GuillaumeGomez) [inline posts/2020-12-18-rustdoc-new-contributors.md:25]: Maybe use links to `docs.rs`?

**Comment by GuillaumeGomez** [comment]
The blog post looks good, thanks a lot! I thin however that we want to hold the publication until https://github.com/rust-lang/rust/pull/79540 is merged (which will **greatly** improve contributions).

File: posts/2020-12-18-rustdoc-new-contributors.md (hunk 1 above)
Comment (apiraino) [reply posts/2020-12-18-rustdoc-new-contributors.md:3]: > "You hate writing docs? So are we! Why not joining rustdoc then?" rofl

this title is pure gold, I hope I'll have a chance in the future to use it :D

**Comment by apiraino** [comment]
Ok, I'm going to close this as it now not necessary anymore (new people joined the team, so things are looking good).

**Comment by jyn514** [comment]
Thank you for writing this up :heart: I'm sorry it never got published, but I really appreciate you caring about it :)

**Pull request closed without merging**